dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java

```diff
@@ -139,44 +140,64 @@ private Identifier getDetailtPageUri(final Structure structure) {
         }
     }
 
-    private boolean containsRegEx(final String uri) {
-        final String mastRegEx = this.getURLMasterPattern().orElse(null);
+    /**
+     * Return all the matches related to a given URI, multiple content types could use the URLMap
+     * pattern and on those cases we need to evaluate all the matches.
+     *
+     * @param uri URI to evaluate for matches
+     * @return List of found matches
+     * @throws DotDataException
+     */
+    private List<Matches> findMatch(final String uri) throws DotDataException {
 
-        if (mastRegEx == null) {
-            return false;
+        // We want to avoid unnecessary lookups for vanity urls when browsing in the backend
+        for (final String backendFilter : CMSUrlUtil.BACKEND_FILTERED_LIST_ARRAY) {
+            if (uri.startsWith(backendFilter)) {
+                return new ArrayList<>();
+            }
         }
 
-        final String url = !uri.endsWith(StringPool.FORWARD_SLASH) ? uri + StringPool.FORWARD_SLASH : uri;
-        return RegEX.contains(url, mastRegEx);
-    }
-
-    private static Optional<String> getURLMasterPattern() {
-        try {
-            final String mastRegEx = CacheLocator.getContentTypeCache().getURLMasterPattern();
-
-            return Optional.ofNullable(mastRegEx);
-        } catch (DotCacheException e) {
-            throw new DotRuntimeException(e);
+        if (this.shouldLoadPatterns()) {
+            this.loadPatterns();
         }
-    }
 
-    private Matches findPatternChange(final String uri) {
-        final String url = !uri.endsWith(StringPool.FORWARD_SLASH) ? uri + StringPool.FORWARD_SLASH : uri;
+        List<Matches> foundMatches = new ArrayList<>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'foundMatches' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5176032)